doc/ipmc/IPMC_Data_Plane_HLD.md

```diff
@@ -0,0 +1,472 @@
+
+#### Warm reboot requirements
+
+This document mainly considers the data plane IP multicast routing function data plane warm reboot:
+
+The warm reboot support for the protocol plane is not in the design scope. In this case, the IP multicast route obtained from the kernel can be regarded as the latest routing data on the data plane.
+
+1. The data plane is warm rebooted
+2. mroutesycnd reads the kernel IP multicast routing information and sends it to the common warm reboot component for comparison
+3. The warm reboot component writes the data to be updated to the APPL DB to complete the IP multicast route update
+4. The warm reboot of the data plane is complete
+
+#### Sequence diagram
+
+Add an IP multicast route (simplify the protocol plane into an FRR component for description. In actual situations, the selection of the protocol plane may be different):
+
+![Add new IP multicast routes](images/add_ipmc_entry.png "Figure 1: Add new IP multicast routes")
+
+Figure 3.  Add new IP multicast routes.
+
+Warm Reboot:
+
+![IP multicast routes warm reboot](images/ipmc_warm_reboot2.png "Figure 1: IP multicast routes warm reboot")
+
+Figure 4. IP multicast routes warm reboot.
+
+#### Scalability and performance requirements
+
+In terms of capacity, the SAI API does not implement a query interface for the capacity of IP multicast routes and multicast member groups like ECMP routes. Therefore, the relevant capacity is not restricted. However, the device must continue to run normally in scenarios where the capacity is exceeded.
+In terms of performance, because the SAI API does not define batch installation of bulk interfaces, IP multicast routes can only be delivered by invoking the SAI API one by one.
+
+### SAI API
+
+The SAI API invocation of IP multicast routing is similar to ECMP routing. A group is created and a member is added. Then, the created rpf group, ipmc group, source IP address, and destination IP address will be bound.
+> Because the SAI API does not yet implement a capacity query interface like ECMP routing, this document is also left blank for IP multicast routing capacity-related processes and interfaces.
+
+In simple terms: 
+> Use sai_ipmc_group_api to create nexthop groups and members
+> Use sai_rpf_group_api to create rpf groups and members
+> Use sai_ipmc_api, source IP address, destination IP address, ipmc group id, and rpf group id to create IP multicast routes
+
+The SAI API involved is as follows:
+| SAI API                                              | SAI attributes |
+| ---------------------------------------------------- | -------------- |
+| sai_router_intfs_api->set_router_interface_attribute | SAI_ROUTER_INTERFACE_ATTR_V4_MCAST_ENABLE</br>SAI_ROUTER_INTERFACE_ATTR_V6_MCAST_ENABLE |
+| sai_rpf_group_api->create_rpf_group                  | NULL           |
+| sai_ipmc_group_api->create_ipmc_group_member         | SAI_RPF_GROUP_MEMBER_ATTR_RPF_GROUP_ID</br>SAI_RPF_GROUP_MEMBER_ATTR_RPF_INTERFACE_ID |

```

Comment:
typo ? it should be sai_rpf_group_api->create_rpf_group_member

Reply:
> typo ? it should be sai_rpf_group_api->create_rpf_group_member

Yes, it will be revised in the next commit.